### ISSUE TITLE
question about input shape

### ISSUE THREAD

**Issue by huangnengCSU** [HawkAaron/warp-transducer#41]
Hi:
    I have a question about the rnnt_loss input shape. The shape of acts is batch x seqLength x (labelLength + 1) x outputDim. However in normal rnn model, the output of rnn network has only three dimension,  which is batch x seqLength x outputDim. Then a linear layer can be used to transform the output shape to batch x seqLength x (labelLength+1). Why here the rnnt_loss input has four dimension.

**Comment by HawkAaron** [comment]
The rnnt model has two components (or three if joint network been used): transcription and prediction network.

The transcription network gets acoustic feature as input, and outputs acoustic hidden representation vectors, shaped [batch, seqLength, outputDim].

The prediction network gets reference label (with blank prepended) as input , and outputs linguistic hidden representation vectors, shaped [batch, labelLength + 1, outputDim].

Finally, the joint network eats both acoustic and linguistic information, outputs a distribution on vocab.

RNN-T has seqLength x (labelLength + 1) number of states, so the logits should be shaped [batch, seqLength, labelLength + 1, vocabDim].

**Comment by huangnengCSU** [comment]
@HawkAaron get it，that is good，thanks a lot.
